nntrainer/graph/network_graph.cpp

```diff
@@ -268,33 +298,43 @@ int NetworkGraph::realizeActivationType(Layer &current) {
   current.output_layers.clear();
   current.output_layers.push_back(layer->getName());
 
-  addLayerNode(layer);
-
   updateNameInLayers(current.getName(), layer->getName());
+  addLayerNode(layer);
```

Comment:
I wonder the order is a matter? layer is the pointer.. so.. It might not be an issue.

Reply:
yes, the order does not matter.
In fact, there is no need for `updateNameInLayers()` here. I will remove it.

Reply:
Turns out there was a bug related to this. Fix added about renaming the connections.